### PR TITLE
feat: notes-context

### PR DESCRIPTION
**Pull request by ankit9015**
Changes added in this PR:
1. Notes Context
2. Save button for Note editor
3. Note Card which displays stored note
4. Notes page is edited to show Pinned Notes and Non pinned notes separately

### CODE REVIEW

File: src/components/LabelButton/LabelButton.jsx [hunk 1]
```diff
@@ -1,12 +1,15 @@
...
 import { MdOutlineLabel } from "../../utils/icons/icons";
 import "./label-button.css";
 
-function LabelButton({ noteData, setNoteData }) {
+function LabelButton() {
   const [newLabel, setNewLabel] = useState("");
   const [showDropdown, setShowDropdown] = useState(false);
-  console.log(noteData);
+  const { notesState, notesDispatch, noteLabels, setNoteLabel } = useNotes();
+  // const isInTags = (label) => notesState.tags.includes(label);
```
Comment (tusharkandpal) [inline src/components/LabelButton/LabelButton.jsx:11]: please remove this commented code.

File: src/components/LabelButton/LabelButton.jsx [hunk 2]
```diff
@@ -32,18 +35,26 @@ function LabelButton({ noteData, setNoteData }) {
...
           </button>
         </div>
+        <ul>
+          {noteLabels.map((label) => (
+            <label key={label}>
+              <input
+                type="checkbox"
+                name={label}
+                // checked={isInTags(label)}
+                // onClick={labelCheckToggle(label)}
```
Comment (tusharkandpal) [inline src/components/LabelButton/LabelButton.jsx:52]: this too.

File: src/context/NotesContext/NotesContext.js [hunk 3]
```diff
@@ -1,13 +1,118 @@
...
+  useState,
+  useEffect,
+  useReducer,
+} from "react";
+import addNoteService from "../../service/Notes/addNoteService";
+
+import getNotesService from "../../service/Notes/getNotesService";
+import {
+  updateNotePinService,
+  updateNoteService,
```
Comment (tusharkandpal) [inline src/context/NotesContext/NotesContext.js:13]: i think one service can do the update note task alone.

File: src/service/Notes/updateNoteService.js [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    console.log(error);
+    error;
+  }
+};
+
+const updateNotePinService = async (currentNote, tokenValue) => {
+  try {
+    const response = await axios.post(
+      `/api/notes/${currentNote._id}`,
+      { note: { ...currentNote, isPinned: !currentNote.isPinned } },
```
Comment (tusharkandpal) [inline src/service/Notes/updateNoteService.js:25]: this `isPinned` you can change before calling this service...which will reduce the repeated code. 
only the `updateNoteService` will do the job.

**Review by tusharkandpal — COMMENTED** [review]
seems good so far.

on visiting hosted site, for me, it seems UI was uneven. Please work on it.

**Review by tusharkandpal — APPROVED** [review]

File: src/components/LabelButton/LabelButton.jsx (hunk 1 above)
Comment (ankit9015) [reply src/components/LabelButton/LabelButton.jsx:11]: it is for labels I tried implementing earlier, will use it in future PR

File: src/service/Notes/updateNoteService.js (hunk 4 above)
Comment (ankit9015) [reply src/service/Notes/updateNoteService.js:25]: This updatepinservice is specific to cards when they are saved, the general updateNoteservice is for editor

**Review by atharvabhanage02 — APPROVED** [review]
Code is well structured and functionalities are working fine , you can merge

**Pull request merged**